x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -65,76 +63,66 @@
 import static org.elasticsearch.xpack.core.security.authc.AuthenticationField.FALLBACK_REALM_TYPE;
 import static org.elasticsearch.xpack.core.security.authc.RealmDomain.REALM_DOMAIN_PARSER;
 
-// TODO(hub-cap) Clean this up after moving User over - This class can re-inherit its field AUTHENTICATION_KEY in AuthenticationField.
-// That interface can be removed
```

Comment:
I don't know what this comment means. But I am pretty sure it no longer applies.

Reply:
It's from when we split `core` and `security` into separate jars. The decision about which class went into each project (and which classes were split down the middle) was made almost entirely on pragmatic grounds rather than a principled design choice. In some cases we split things apart just to make it work (a day or so before feature freeze).

Reply:
Thanks for the context. So the removal is safe enough.